modules/cnf-topology-aware-lifecycle-manager-backup-feature.adoc

```diff
@@ -0,0 +1,160 @@
+
+For SNO (single node cluster), you can create a backup of a deployment before an upgrade. If the upgrade fails you can use the backup to return the system to its original state.
+The backup consists of:
+
+Cluster backup:: A snapshot of etcd and static pod manifests.
+Content backup:: Backups of folders, for example, `/etc`, `/usr/local`, `/var/lib/kubelet`.
+Changed files backup:: Any file managed by `machine-config` that has been changed.
+Deployment:: A pinned `ostree` deployment.
+Images (Optional):: Any container images that are in use.
+
+
+.Prerequisites
+
+* Install the {cgu-operator-first}.
+* Provision one or more managed clusters.
+* Log in as a user with `cluster-admin` privileges.
+* Install Advanced Cluster Management
+
+[NOTE]
+====
+It is highly recommended that you create a recovery partition.
+The following is an example `SiteConfig` CR for a recovery partition of 50Gb:
+
+[source,yaml]
+----
+nodes:
+    - hostName: "snonode.sno-worker-0.e2e.bos.redhat.com"
+    role: "master"
+    rootDeviceHints:
+        hctl: "0:2:0:0"
+        deviceName: /dev/sda
+........
+........
+    #Disk /dev/sda: 893.3 GiB, 959119884288 bytes, 1873281024 sectors
+    diskPartition:
+        - device: /dev/sda
+        partitions:
+        - mount_point: /var/recovery
+            size: 51200
+            start: 800000
+----
+====
+
+.Procedure
+
+. Save the contents of the `ClusterGroupUpgrade` CR with the `backup` field set to `true` in the `clustergroupupgrades-group-du.yaml` file:
++
+[source,yaml]
+----
+apiVersion: ran.openshift.io/v1alpha1
+kind: ClusterGroupUpgrade
+metadata:
+  name: du-upgrade-4918
+  namespace: ztp-group-du-sno
+spec:
+  preCaching: true
+  deleteObjectsOnCompletion: true
+  backup: true
+  clusters:
+  - cnfdb1
+  - cnfdb2
+  enable: false
+  managedPolicies:
+  - du-upgrade-platform-upgrade
+  remediationStrategy:
+    maxConcurrency: 2
+    timeout: 240
+----
+
+. To start the update, apply the `ClusterGroupUpgrade` CR by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f clustergroupupgrades-group-du.yaml
+----
+
+.Verification
+
+. Check the status of the upgrade in the hub cluster by running the following command:
++
+[source,terminal]
+----
+$ oc get cgu -n ztp-group-du-sno du-upgrade-4918 -o jsonpath='{.status}'
+----
++
+.Example output
++
+[source,json]
+----
+{
+    "backup": {
+        "clusters": [
+            "test-spoke2",
+            "test-spoke1"
+    ],
+    "status": {
+        "test-spoke1": "Succeeded",
+        "test-spoke2": "Succeeded"
+    }
+},
+"computedMaxConcurrency": 1,
+"conditions": [
+    {
+        "lastTransitionTime": "2022-04-05T10:37:19Z",
+        "message": "Backup is completed",
+        "reason": "BackupCompleted",
+        "status": "True",
+        "type": "BackupDone"
+    }
+],
+"precaching": {
+    "spec": {}
+},
+"status": {}
+----
+
+. Check the status of the upgrade by running the following command:
```

Comment:
Backup and recovery should not be in the same place, recovery is not part of the backup. recovery is only done when a upgrade actually fails. To discuss.

Reply:
Agreed - added new section